src/main/java/com/rabbitmq/http/client/ReactorNettyClient.java

```diff
@@ -0,0 +1,170 @@
+import io.netty.handler.codec.json.JsonObjectDecoder;
+import org.reactivestreams.Publisher;
+import org.springframework.web.util.UriComponentsBuilder;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.ipc.netty.http.client.HttpClient;
+import reactor.ipc.netty.http.client.HttpClientRequest;
+import reactor.ipc.netty.http.client.HttpClientResponse;
+
+import java.lang.reflect.Array;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.function.Function;
+
+/**
+ *
+ */
+public class ReactorNettyClient {
+
+    private static final int MAX_PAYLOAD_SIZE = 100 * 1024 * 1024;
+
+    private final Mono<String> root = Mono.just("http://localhost:15672/api");
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    private final HttpClient client;
+
+    private final String authorizationHeader;
+
+    public ReactorNettyClient() {
+        // FIXME make Jackson ObjectMapper configurable
+        objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        objectMapper.enable(DeserializationFeature.ACCEPT_EMPTY_ARRAY_AS_NULL_OBJECT);
+        objectMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+        objectMapper.disable(MapperFeature.DEFAULT_VIEW_INCLUSION);
+
+        // FIXME make URL configurable
+        UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromHttpUrl(root.block());
+        client = HttpClient.create(options -> options.host(uriBuilder.build().getHost()).port(uriBuilder.build().getPort()));
+
+        // FIXME make Authentication header value configurable (default being Basic)
+        String credentials = "guest" + ":" + "guest";
+        byte[] credentialsAsBytes = credentials.getBytes(StandardCharsets.ISO_8859_1);
+        byte[] encodedBytes = Base64.getEncoder().encode(credentialsAsBytes);
+        String encodedCredentials = new String(encodedBytes, StandardCharsets.ISO_8859_1);
+        authorizationHeader = "Basic " + encodedCredentials;
+
+        // FIXME make SSLContext configurable when using TLS
+    }
+
+    public Mono<OverviewResponse> getOverview() {
+        return doGetMono(builder -> builder.pathSegment("overview"), OverviewResponse.class);
+    }
+
+    public Flux<NodeInfo> getNodes() {
+        return doGetFlux(builder -> builder.pathSegment("nodes"), NodeInfo.class);
+    }
+
+    public Mono<HttpClientResponse> declarePolicy(String vhost, String name, PolicyInfo info) {
+        return doPost(builder -> builder.pathSegment("policies", vhost, name), info);
+    }
+
+    private HttpClientRequest disableChunkTransfer(HttpClientRequest request) {
+        return request.chunkedTransfer(false);
+    }
+
+    public Flux<PolicyInfo> getPolicies() {
+        return doGetFlux(builder -> builder.pathSegment("policies"), PolicyInfo.class);
+    }
+
+    public Mono<HttpClientResponse> deletePolicy(String vhost, String name) {
```

Comment:
Does sounds like `HttpClientResponse` is a high-level API for REST.
Since we talk here only about `DELETE` wouldn't it be just enough to return a status instead of the whole low-level `HttpClientResponse`?

Thanks

Reply:
Either that or a boolean (whether the delete succeeded).

Reply:
Good question. This applies to empty responses (POST, PUT, and DELETE). Different options:
 * stick to `HttpClientResponse`: simplest, complete, but leaky
 * create our own HTTP response wrapper: more involving but the most flexible
 * return the bare minimal (boolean or status code): simple, but limiting

I'd go with the wrapper, with only the status code and headers, we could make it more elaborate on demand.